File: 15 exercicios POO/6/python/motor.py
class Motor:
    def __init__(self, tipo, potencia):
        self.tipo = tipo
        self.potencia = potencia
    def ligar(self):
        print("O motor está ligado.")
    def desligar(self):
        print("O motor está desligado.")

class Carro:
    def __init__(self, marca, modelo):
        self.marca = marca
        self.modelo = modelo
        self.motor = Motor("Gasolina", 150)
    def ligar(self):
        self.motor.ligar()
        print("O carro está pronto para rodar.")
    def desligar(self):
        self.motor.desligar()
        print("O carro foi desligado.")

carro = Carro("Toyora", "corolla")
carro.ligar()
carro.desligar()

File: 15 exercicios POO/6/java/src/Motor.java
public class Motor {
    public String getTipo() {
        return tipo;
    }

    private String tipo;
    private int potencia;

    public Motor(String tipo, int potencia){
        this.tipo = tipo;
        this.potencia = potencia;
    }

    public void ligar() {
        System.out.println("O motor está ligado.");
    }

    public void desligar() {
        System.out.println("O motor está desligado.");
    }
}
